lib/octopus_serverspec_extensions/type/octopus_deploy_worker.rb

```diff
@@ -0,0 +1,173 @@
+      @worker["SpaceId"] == space_id
+    end
+
+    def has_policy?(policy_name)
+      return false if @worker.nil?
+      url = "#{@serverUrl}/api/#{@spaceFragment}machinepolicies/all?api-key=#{@apiKey}"
+      resp = Net::HTTP.get_response(URI.parse(url))
+      policies = JSON.parse(resp.body)
+      policy_id = policies.select {|e| e["Name"] == policy_name}.first["Id"]
+      @worker["MachinePolicyId"] == policy_id
+    end
+
+    def has_display_name?(name)
+      return false if @worker.nil?
+      @worker["Name"] == name
+    end
+
+    def has_endpoint?(uri)
+      return false if @worker.nil?
+      return false if @worker["Uri"].nil? # polling tentacles have null endpoint. catch that.
+      puts "Expected uri '#{uri}' for Worker #{@name}, but got '#{@worker["Uri"]}'" unless (@worker["Uri"].casecmp(uri) == 0)
+      @worker["Uri"].casecmp(uri) == 0
+    end
+
+    def listening_tentacle?
+      return false if @worker.nil?
+      puts "Expected CommunicationStyle 'TentaclePassive' for Tentacle #{@name}, but got '#{@worker["Endpoint"]["CommunicationStyle"]}'" if (@worker["Endpoint"]["CommunicationStyle"] != "TentaclePassive")
+      @worker["Endpoint"]["CommunicationStyle"] == "TentaclePassive"
+    end
+
+    def polling_tentacle?
+      return false if @worker.nil?
+      puts "Expected CommunicationStyle 'TentacleActive' for Tentacle #{@name}, but got '#{@worker["Endpoint"]["CommunicationStyle"]}'" if (@worker["Endpoint"]["CommunicationStyle"] != "TentacleActive")
+      @worker["Endpoint"]["CommunicationStyle"] == "TentacleActive"
+    end
+
+    def exists?
+      ::File.exists?("c:\\program files\\Octopus Deploy\\Tentacle\\Tentacle.exe")
+    end
+  end
+
+  def octopus_deploy_worker(serverUrl, apiKey, instance)
+    OctopusDeployWorker.new(serverUrl, apiKey, instance)
+  end
+
+  private
+
+  def check_supports_spaces(serverUrl)
+    begin
+      resp = Net::HTTP.get_response(URI.parse("#{serverUrl}/api/"))
+      body = JSON.parse(resp.body)
+      version = body['Version']
+      return Gem::Version.new(version) > Gem::Version.new('2019.0.0')
+    rescue => e
+      puts "Unable to connect to #{serverUrl}: #{e}"
+    end
+
+    return false
+  end
+
+  def poll_until_worker_has_completed_healthcheck(serverUrl, apiKey, thumbprint)
+    worker = nil
+    url = "#{serverUrl}/api/#{@spaceFragment}workers/all?api-key=#{apiKey}"
+
+    now = Time.now
+    counter = 1
+    loop do
+      worker = get_worker_via_api(serverUrl, apiKey, thumbprint)
+
+      break if worker.nil?
+      break if counter > 10
+      break if !worker_healthcheck_outstanding(worker)
+      puts "Machine health check for #{worker["Name"]} has not yet completed. Waiting 5 seconds to try again."
+      counter += 1
+      sleep 5
+    end
+
+    worker
+  end
+
+  def worker_healthcheck_outstanding(worker)
+    worker["StatusSummary"] == "This worker was recently added. Please perform a health check."
```

Comment:
is this the actual error message?

Reply:
Good question. I should check.

Reply:
It's "machine was recently". Flipping that back.